### DISCUSSION TITLE
安装好了，其他功能都正常，就是使用打包文件下载的功能异常

### DISCUSSION THREAD

**Discussion by samero** [AlistGo/alist#393] [Q&A] [answered]
安装好了，其他功能都正常，就是使用打包文件下载的功能异常，换过很多浏览器，选择文件夹下载或者打包多个文件下载的时候，只会弹出一个小窗口，路径如下 http://XXXXX:5244/public/mitm.html ,然后目录下面提示“”正在下载文件，请勿关闭或刷新页面”，但等了很久也没有出现下载，有时候打包的文件含有 .exe 文件，它会下载 .exe文件，其他文件都不会被打包下载，请问这是什么原因，是需要centos7系统安装什么支持的软件吗？

**Comment by xhofe** [comment, marked as answer]
> Use https if you can. That way you don't have to open the man in the middle in a popup to install the service worker from another secure context. Popups are often blocked but if you can't it's best that you initiate the createWriteStream on user interaction. Even if you don't have any data ready - this is so that you can get around the popup blockers. (In secure context this don't matter) Another benefit of using https is that the mitm-iframe can ping the service worker to prevent it from going idle. (worker goes idle after 30 sec in firefox, 5 minutes in blink) but also this won't mater if the browser supports transferable streams throught postMessage since service worker don't have to handle any logic. (the stream that you transfer to the service worker will be the stream we respond with)
>
> Handle unload event when user leaves the page. The download gets broken when you leave the page. Because it looks like a regular native download process some might think that it's okey to leave the page beforehand since it's is downloading in the background directly from some a server, but it isn't.

打包下载需要使用https，因为它需要使用Service Workers

**Reply by samero** [reply]
使用https后果然可以正常打包下载，谢谢了
